### PR TITLE
test: add edge-case tests for api_key_prefix

### PR DESCRIPTION
**Pull request by Srishti-Gupta74**
## Summary

* Added a new `tests/test_auth.py` file with dedicated edge-case tests for `api_key_prefix`.
* Covered all scenarios listed in Issue #254, including standard API keys, multiple dots, short and long keys without dots, exact 16-character keys, whitespace handling, empty inputs, dot-only inputs, and the length-bound property assertion.
* This was needed because `api_key_prefix` is used in security-relevant logging and audit paths but previously had no dedicated edge-case test coverage.

## Testing

* [x] `WAGGLE_MODEL=deterministic pytest -q`
* [x] `ruff check src/ tests/`
* [x] `ruff format --check src/ tests/`

Additional verification:

* [x] `python -m pytest tests/test_auth.py -q`
* [x] `python -m ruff check tests/test_auth.py`

### Test report

```text
$ python -m pytest tests/test_auth.py -q

..........                                                               [100%]
10 passed in 2.50s

$ WAGGLE_MODEL=deterministic pytest -q

608 passed, 5 skipped, 1 warning in 131.69s
```

Notes:

* `ruff check src/ tests/` passed locally.
* The full test suite passed locally after installing the project's development dependencies.

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage for authentication key validation with comprehensive unit tests covering various input scenarios and edge cases.

### CODE REVIEW

File: tests/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        "",
+        "a",
+        "short",
+        "a" * 16,
+        "a" * 32,
+        "a" * 100,
+    ]
+
+    for sample in samples:
+        assert len(api_key_prefix(sample)) <= 16
```
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply tests/test_auth.py:52]: Thanks for the suggestion. For this PR I followed the issue's acceptance criteria, which specifically calls out the 16-character bound for inputs without a dot.

The current implementation returns everything before the first dot, so inputs such as `"a" * 20 + ".secret"` intentionally produce a prefix longer than 16 characters. Since this PR is focused on adding test coverage for the existing behavior rather than changing it, I limited the property-style assertion to the no-dot case described in the issue.

If the dot-present behavior should also be capped at 16 characters, that would be a separate behavioral change and may be better handled in a follow-up issue/PR.

**Comment by Srishti-Gupta74** [comment]
Hi @ard12, @Abhigyan-Shekhar!
While working on these tests, I noticed that the 16-character limit only applies when no dot is present. For example, a very long prefix before the first dot is currently returned in full.
I didn't change that behavior because this PR is focused on test coverage for the existing implementation, but I'm happy to open a follow-up issue if you'd like it tracked.

File: tests/test_auth.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Srishti-Gupta74) [reply tests/test_auth.py:52]: Thanks for the suggestion. I kept this PR scoped to the issue's acceptance criteria and existing behavior. If maintainers agree this should be tracked separately, I would be happy to open a follow-up issue for the dot-present case.

**Review by Abhigyan-Shekhar — APPROVED** [review]
Maintainer pass on June 11, 2026: scope is tight, the earlier thread is resolved, and the GitHub CI matrix is green.

**Comment by Srishti-Gupta74** [comment]
Hi @Abhigyan-Shekhar! Thanks for the review and clarification!
